File: backend/app/models/evolution.py
from typing import Optional

from pydantic import BaseModel


class MetricPointResponse(BaseModel):
    metric: str
    player_value: float
    position_average: Optional[float]
    percentile: Optional[float]
    trend: Optional[str]


class SeasonEvolutionResponse(BaseModel):
    season: str
    metrics: list[MetricPointResponse]


class PlayerEvolutionResponse(BaseModel):
    player_id: int
    name: str
    position: str
    sub_position: str
    available_metrics: list[str]
    seasons: list[SeasonEvolutionResponse]

File: frontend/src/types/evolution.ts
export type Trend = "improved" | "declined" | "stable";

export interface MetricPoint {
  metric: string;
  player_value: number;
  position_average: number | null;
  percentile: number | null;
  trend: Trend | null;
}

export interface SeasonEvolution {
  season: string;
  metrics: MetricPoint[];
}

export interface PlayerEvolutionResponse {
  player_id: number;
  name: string;
  position: string;
  sub_position: string;
  available_metrics: string[];
  seasons: SeasonEvolution[];
}
